docs/data-sources/packer_image_iteration.md

```diff
@@ -0,0 +1,69 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "hcp_packer_image_iteration Data Source - terraform-provider-hcp"
+subcategory: ""
+description: |-
+  The Packer Image data source iteration gets the most recent iteration (or build) of an image, given a channel.
+---
+
+# hcp_packer_image_iteration (Data Source)
+
+The Packer Image data source iteration gets the most recent iteration (or build) of an image, given a channel.
+
+
```

Comment:
This is auto generated but, can I edit things here ?

Reply:
To edit the generated doc, you'll want to add a `packer_image_iteration` template to https://github.com/hashicorp/terraform-provider-hcp/tree/main/templates/data-sources. You can follow the example of HVN route, and add whatever you need.

Follow-up question, is this feature generally available? If not, you'll want to add a banner in the template: `-> **Note:** This feature is currently in private/public beta.`

Reply:
It's private beta ! I added that, and tweaked things accordingly 👍🏼  thanks  !

Reply:
is there a way to preview docs ?

Reply:
Also, some nested attributes — like the `builds` one — are no documented, even though I did document the fields in the provider code; is there something I can do to fix that ?

Reply:
You can use this tool to preview docs, just have to copy-paste the markdown into the form: https://registry.terraform.io/tools/doc-preview

I'll check to see if the doc generator supports doubly nested objects. I'm thinking you'll need to add the `builds` documentation manually to the template in the meantime

Reply:
> You can use this tool to preview docs, just have to copy-paste the markdown into the form: https://registry.terraform.io/tools/doc-preview

Dope, thanks ! Sorry I missed that info.

> The nested object fields get displayed below in a Nested Schema section:

Nice, also, I documented them in here:

https://github.com/hashicorp/terraform-provider-hcp/blob/7ae2cd98f817082540adc8234acef46f86e00c76/internal/provider/data_source_packer_image_iteration.go#L61-L76

But they are not documented here:

https://github.com/hashicorp/terraform-provider-hcp/blob/670702418fd8ac0837208c18d8a6f20a82b5cfc9/docs/data-sources/packer_image_iteration.md#L52-L59

Reply:
So, I looked and long story short, because of the way TF schema works, this is not possible.